packages/lodestar/src/api/impl/beacon/blocks/index.ts

```diff
@@ -147,7 +149,18 @@ export function getBeaconBlockApi({
 
       metrics?.registerBeaconBlock(OpSource.api, seenTimestampSec, signedBlock.message);
 
-      await Promise.all([chain.receiveBlock(signedBlock), network.gossip.publishBeaconBlock(signedBlock)]);
+      await Promise.all([
+        // Send the block, regardless of whether or not it is valid. The API
+        // specification is very clear that this is the desired behaviour.
+        network.gossip.publishBeaconBlock(signedBlock),
+
+        chain.processBlock(signedBlock).catch((e) => {
+          if (e instanceof BlockError && e.type.code === BlockErrorCode.PARENT_UNKNOWN) {
```

Comment:
this does not happen if a validator publish a block produced by us. Is this to protect against an attacker trying to publish random blocks? How about using CSRF token for all of our apis?

Reply:
nvm, CSRF token should not be used for lodestar only as it should be part of the spec